File: frontend/lib/models/shift_model.dart
class Shift {
  final String id;
  final String name;
  final String startTime;
  final String endTime;
  final String punchInRule;
  final String punchOutRule;
  final int punchInHours;
  final int punchInMinutes;
  final int punchOutHours;
  final int punchOutMinutes;

  Shift({
    required this.id,
    required this.name,
    required this.startTime,
    required this.endTime,
    this.punchInRule = 'Anytime',
    this.punchOutRule = 'Anytime',
    this.punchInHours = 0,
    this.punchInMinutes = 0,
    this.punchOutHours = 0,
    this.punchOutMinutes = 0,
  });

  factory Shift.fromJson(Map<String, dynamic> json) {
    return Shift(
      id: json['_id'] ?? json['id'] ?? '',
      name: json['name'] ?? '',
      startTime: json['startTime'] ?? '09:00 AM',
      endTime: json['endTime'] ?? '06:00 PM',
      punchInRule: json['punchInRule'] ?? 'Anytime',
      punchOutRule: json['punchOutRule'] ?? 'Anytime',
      punchInHours: json['punchInHours'] ?? 0,
      punchInMinutes: json['punchInMinutes'] ?? 0,
      punchOutHours: json['punchOutHours'] ?? 0,
      punchOutMinutes: json['punchOutMinutes'] ?? 0,
    );
  }
}

File: backend/models/shiftModel.js
// models/shiftModel.js
import mongoose from 'mongoose';

const shiftSchema = new mongoose.Schema(
  {
    company: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Company',
      required: true,
    },
    name: {
      type: String,
      required: true,
      trim: true,
    },
    // store as "HH:MM AM/PM" like your UI,
    // or change to Date/Number if you prefer
    startTime: {
      type: String,
      required: true,
    },
    endTime: {
      type: String,
      required: true,
    },
    punchInRule: {
        type: String,
        enum: ['Anytime', 'Add Limit'],
        default: 'Anytime',
      },
      
      punchOutRule: {
        type: String,
        enum: ['Anytime', 'Add Limit'],
        default: 'Anytime',
      },
    // optional: numeric limits if "Add Limit" is used
    punchInHours: {
      type: Number,
      default: 0,
    },
    punchInMinutes: {
      type: Number,
      default: 0,
    },
    punchOutHours: {
      type: Number,
      default: 0,
    },
    punchOutMinutes: {
      type: Number,
      default: 0,
    },
    isActive: {
      type: Boolean,
      default: true,
    },
  },
  { timestamps: true }
);

const Shift = mongoose.model('Shift', shiftSchema);
export default Shift;
